vunit/project.py

```diff
@@ -433,6 +434,20 @@ def comparison_key(source_file):
         sorted_files = sorted(affected_files, key=comparison_key)
         return sorted_files
 
+    def _get_affected_files(self, target_files, get_depend_func):
```

Comment:
It seems that lines 384-396 and 423-435 of the resulting `project.py` file are redundant. Does it make sense to move those snippets into this new function (maybe renamed to `_get_affected_files_in_compile_order`)?

Reply:
Then I have to pass dependency_graph and dependeny_graph.dependent/dependencies or I write a closure function passing dependent or dependencies returning the function you propose. Do you have a preference?

Reply:
Although I might have a preferene, I think it is not relevant here. It was a somehow 
rhetorical question: are you going to later (in some other PR) use any of the three individual functions from somewhere else in the codebase and/or do you want to expose any of them? If the answer to any of these is yes, keeping them separated is preferred/required. However, should it be an *atomic* chain (external processes always execute A, then B, then C), the creation of three functions might be better implemented as three comments/sections in a single function. Nevertheless, we are not doing embedded programming here, so having three function calls is irrelevant from a performance perspective.

Reply:
Originally I did use the three functions individually. But now in my final solution I actually do not. So I would agree that it could be preferable to have one function. On the other hand the three functions actually have three 'responsibilities' and therefor I personally find it easier to understand when reading the code. 

I completely agree with you and find the discussion very interesting. My background is also more an embedded one, but here I was actually trying to follow Uncle Bob's advice from 'Clean Code'.

Alternatively it would also be possible to write one function which orchestrates (wraps) these three functions so that there is only one function call in get_files_in_compile_order and get_dependencies_in_compile_order. Again this is not ideal from performance perspective, but would keep the functions small which makes it easier to understand the intent.

Reply:
I believe that this discussion we had is enough for anyone considering to write the wrapper/orchestration function in the future. Thanks!

It is not required for now, and you/we would better focus on further features/contributions. Just resolve the conflicts, and we can move forward.

Reply:
Clean code is a good book to use as a guidance. Small function with a clearly defined responsibility is worth striving for.